### PR TITLE
move NixOS VM tests to `checks` output

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
this should capture all the tests to be run by buildbot

it also effectively deprecates Hydra builds

paired with @erethon; please run the formatter (development shell doesn't work on Darwin)

### CODE REVIEW

**Comment by fricklerhandwerk** [comment]
@mightyiam the mCaptcha VM test fails, and you were the last one to touch the code.

**Comment by stepbrobd** [comment]
Do you intend to rebase this or would you like us (@ngi-nix/magic @ngi-nix/cake) to do it?

**Comment by fricklerhandwerk** [comment]
@stepbrobd thanks, I'll manage :)

**Comment by fricklerhandwerk** [comment]
@Erethon @lorenzleutgeb by the way, the new CI is amazing. Pure refactorings like these finish checks in < 1min.

File: flake.nix [hunk 1]
```diff
@@ -271,30 +272,32 @@
...
-          nixos = mapAttrByPath ["nixos" "tests"] {} nonBrokenNgiProjects;
-        };
-
-        nixosConfigurations.${system} =
+      # See ./infra/makemake/buildbot.nix for how it's set up.
+      # NOTE: `nix flake check` requires a flat attribute set of derivations, which is an annoying constraint...
+      checks.${system} = let
+        vmTests = name: project:
+          optionalAttrs (project ? nixos.tests)
+          (concatMapAttrs (testname: test: {"projects/${name}/passthru/${testname}" = test;}) project.nixos.tests);
```
Comment (fricklerhandwerk) [inline flake.nix:280]:
```suggestion
          (concatMapAttrs (testname: test: {"projects/${name}/vmTest/${testname}" = test;}) project.nixos.tests);
```

**Review by lorenzleutgeb — APPROVED** [review]
LGTM to me now that I refactored it, lol. Passing the ball back to @fricklerhandwerk for further review or merge.

File: flake.nix [hunk 2]
```diff
@@ -221,8 +228,35 @@
...
+          {"packages/${packageName}" = package;}
+          // (concatMapAttrs (passthruName: test: {"packages/${packageName}/passthru/${passthruName}" = test;}) (package.passthru.tests or {}));
+
+        checksForAllPackages =
+          concatMapAttrs
+          checksForPackage
+          nonBrokenNgiPackages;
+      in
+        checksForAllPackages
+        // checksForAllPackages
```
Comment (fricklerhandwerk) [inline flake.nix:259]:
```suggestion
```

Artifact?

File: flake.nix [hunk 3]
```diff
@@ -110,10 +109,11 @@
         sops-nix = sops-nix.nixosModules.default;
       };
 
-    extendedNixosConfigurations =
-      mapAttrs
-      (_: config: nixosSystem {modules = [config ./dummy.nix] ++ attrValues extendedNixosModules;})
-      rawExamples;
+    mkNixosSystem = config: nixosSystem {modules = [config ./dummy.nix] ++ attrValues extendedNixosModules;};
```
Comment (fricklerhandwerk) [inline flake.nix:112]: For another time: I'm not sure what this `dummy` configuration is for, it seems like it's a remnant of past times and can go away entirely.

Comment (lorenzleutgeb) [reply flake.nix:112]: Can we please discuss this in a separate issue. The code is moved here, but it's out of scope for this PR.

File: flake.nix (hunk 2 above)
Comment (lorenzleutgeb) [reply flake.nix:259]: Bug. Missing projects. Fixing.

File: flake.nix (hunk 3 above)
Comment (lorenzleutgeb) [reply flake.nix:112]: Deferring to #275

**Comment by mightyiam** [comment]
Thank you!

**Pull request merged**